src/utils/errors.ts

```diff
@@ -397,3 +397,42 @@ export class UnexpectedError extends CoreError {
     this.origin = 'unknown';
   }
 }
+
+export class ShortLinkNotFound extends CoreError {
+  constructor(shortLink: string) {
+    super(
+      {
+        code: 'GERR1009',
+        statusCode: StatusCodes.NOT_FOUND,
+        message: FAILURE_MESSAGES.SHORT_LINK_NOT_FOUND,
+      },
+      shortLink,
+    );
+  }
+}
+
+export class ShortLinkDuplication extends CoreError {
+  constructor(shortLink: string) {
+    super(
+      {
+        code: 'GERR1010',
```

Comment:
Actually create a new "errors" file and extend the error from another factory. So the origin can be short link instead of unknown. There are examples in the repo.